EdenGri/sonic-swss
# Lab5 - add tx error monitor

**What I did**

Add new orch agent txmonitororch. the txmonitor orch monitor the tx errors on switch ports. In case the number of  errors on one port in a configurable time period reaches the configurable thresholds , port is deemed as “Not OK”. 

**Why I did it**

In order to get the status of the tx for each port in the switch.

**How I verified it**

enter the following CLI commands:
test 1:
1. show tx-error-monitor status 
2. choose one oid of port with the status "ok"
3. stop the counters:
    counterpoll port disable
4. set the tx counter of the port to be above threshold:
     redis-cli -n 2 HSET COUNTERS:oid:<oid> SAI_PORT_STAT_IF_OUT_ERRORS "200"
5. verify that the status of the port change to not ok: 
     show tx-error-monitor status
test 2:
1. show tx-error-monitor status 
2. choose one oid of port with the status "ok"
3. stop the counters:
    counterpoll port disable
4. set the threshold to 200:
    config tx-config threshold 200
6. set the tx counter of the port to be above threshold:
     redis-cli -n 2 HSET COUNTERS:oid:<oid> SAI_PORT_STAT_IF_OUT_ERRORS "200"
7. verify that the status of the port is still ok: 
     show tx-error-monitor status

**Details if related**
